src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.ts

```diff
@@ -452,6 +471,76 @@ export class InstructorSessionEditPageComponent extends InstructorSessionBasePag
     });
   }
 
+  handlePromptDeletionOfDeadlines(submissionEndTimestamp: number): Observable<boolean> {
+    const hasDeadlinesBeforeUpdatedEndTime = DeadlineExtensionHelper
+      .hasDeadlinesBeforeUpdatedEndTime(this.studentDeadlines, this.instructorDeadlines, submissionEndTimestamp);
+
+    if (!hasDeadlinesBeforeUpdatedEndTime) {
+      return of(true); // no need to prompt for deletion
+    }
+
+    const [affectedStudentDeadlines, affectedInstructorDeadlines] = this
+      .getAffectedIndividualDeadlines(submissionEndTimestamp);
+    const [affectedStudentModels, affectedInstructorModels] = this
+      .getAffectedIndividualModels(submissionEndTimestamp, affectedStudentDeadlines, affectedInstructorDeadlines);
+
+    const modalRef: NgbModalRef = this.ngbModal.open(ExtensionConfirmModalComponent);
+    modalRef.componentInstance.modalType = ExtensionModalType.VALIDATE_DEADLINE;
+    modalRef.componentInstance.selectedStudents = affectedStudentModels;
+    modalRef.componentInstance.selectedInstructors = affectedInstructorModels;
+    modalRef.componentInstance.extensionTimestamp = submissionEndTimestamp;
+    modalRef.componentInstance.feedbackSessionTimeZone = this.sessionEditFormModel.timeZone;
+
+    return new Observable((subscribeIsUserAccept) => {
+      modalRef.componentInstance.onConfirmExtensionCallBack.subscribe(() => {
+        this.updateDeadlines(affectedStudentModels, affectedInstructorModels, affectedStudentDeadlines,
+          affectedInstructorDeadlines);
+        modalRef.componentInstance.isSubmitting = false;
+        modalRef.close();
+        subscribeIsUserAccept.next(true);
+      }, () => {
+        subscribeIsUserAccept.next(false);
+      });
+    });
+  }
+
+  private getAffectedIndividualDeadlines(submissionEndTimestamp: number): [
+    Record<string, number>, Record<string, number>,
+  ] {
+    const affectedStudentDeadlines = DeadlineExtensionHelper.setDeadlinesBeforeEndTime(
+      this.studentDeadlines, submissionEndTimestamp);
+    const affectedInstructorDeadlines = DeadlineExtensionHelper.setDeadlinesBeforeEndTime(
+      this.instructorDeadlines, submissionEndTimestamp);
+    return [affectedStudentDeadlines, affectedInstructorDeadlines];
+  }
+
+  private getAffectedIndividualModels(
```

Comment:
nit: Could we add a comment to this method. "Gets individual deadlines that ...". I suspect someone who is not familiar with the context might be wondering what "affected" means.